File: mobileapp/lib/models/invitation_model.dart
class InvitationModel {
  final int id;
  final int? projectId;
  final String projectName;
  final String inviteeEmail;
  final int? invitedById;
  final String invitedByUsername;
  final String message;
  final String? expiresAt;
  final bool accepted;
  final String createdAt;

  const InvitationModel({
    required this.id,
    this.projectId,
    required this.projectName,
    required this.inviteeEmail,
    this.invitedById,
    required this.invitedByUsername,
    required this.message,
    this.expiresAt,
    required this.accepted,
    required this.createdAt,
  });

  bool get isPending => !accepted;
  String get status => accepted ? 'accepted' : 'pending';

  factory InvitationModel.fromJson(Map<String, dynamic> json) {
    final acceptedBool = json['accepted'] as bool? ?? false;
    final projectRaw = json['project'];
    final inviterRaw = json['inviter'];

    return InvitationModel(
      id: json['id'] as int,
      projectId: _extractId(projectRaw),
      projectName: json['project_name'] as String? ?? 'Projekt',
      inviteeEmail: json['email'] as String? ?? '',
      invitedById: _extractId(inviterRaw),
      invitedByUsername: json['inviter_username'] as String? ?? '',
      message: json['message'] as String? ?? '',
      expiresAt: json['expires_at'] as String?,
      accepted: acceptedBool,
      createdAt: json['created_at'] as String? ?? '',
    );
  }

  static int? _extractId(dynamic value) {
    if (value == null) return null;
    if (value is int) return value;
    if (value is Map) return value['id'] as int?;
    if (value is String) {
      final parts = value.split('/').where((s) => s.isNotEmpty).toList();
      return parts.isNotEmpty ? int.tryParse(parts.last) : null;
    }
    return null;
  }
}

File: desktopapp/desktopapp/Models/InvitationModel.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;

namespace desktopapp.Models
{
    public class InvitationModel
    {
        [Key]
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("email")]
        public string? Email { get; set; }
        
        [JsonProperty("inviter")]
        public int InviterId { get; set; }
        
        [JsonProperty("project")]
        public int ProjectId { get; set; }
        
        [JsonProperty("project_name")]
        public string? ProjectName { get; set; }

        [JsonProperty("inviter_username")]
        public string? InviterUsername { get; set; }

        [JsonProperty("message")]
        public string? Message { get; set; }

        [JsonProperty("created_at")]
        public System.DateTime CreatedAt { get; set; }
        
        [JsonProperty("expires_at")]
        public System.DateTime? ExpiresAt { get; set; }

        [JsonProperty("accepted")]
        public bool Accepted { get; set; }
        
        public string DisplayProject => !string.IsNullOrEmpty(ProjectName) ? ProjectName : "Nieznany Projekt";
        
        public string DisplayInviter => !string.IsNullOrEmpty(InviterUsername) ? InviterUsername : "Nieznany Zapraszający";
    }
}